### ISSUE TITLE
Salesforce CLI Extension last version not working on Mac M1

### ISSUE THREAD

**Issue by martinlezer** [forcedotcom/salesforcedx-vscode#4674]
### Summary

Impossible to retrieve or push sources using the CLI extension since the last version

### Steps To Reproduce:

1.  Authorize an org with a sandbox
1.  Try to retrieve source from a package.xml or from the from a folder or from the org browser

### Expected result

Sources should be retrieved

### Actual result

The command runs forever without returning any result

### Additional information

I am on Mac M1 with the latest versions of all Salesforce extensions.
When I switch the CLI extension to the previous version, everything works.
When I use sfdx through the terminal, it also works.

**Salesforce Extension Version in VS Code**: 56.16.0

**SFDX CLI Version**: 56.16.0

**OS and version**: Mac OS Ventura 13.0.1

**Comment by gbockus-sf** [comment]
Hi @martinlezer 
Thanks for reporting this issue.  
We tried to repo this without success.  I suspect you were being impacted by a caching issue we've been fighting around authorizations. 

A couple of questions for you:
Did you authorize the org for the first time in the project (open the browser login etc)? Or did you select the already authorized org from the list in the org launcher? 

When you get a second can you try switching back to v56.16 and attempting the retrieve again (I mostly want to verify that after a reload from updating the core extension that the deploy is still hanging).  

If it does still hang can you then try running from the command palette: 'Developer: Reload Window' and try again after the extensions loads (hopefully the authorized or will already be selected)?  (Also let it sit for a while...just to verify that it doesn't eventually resolve).

Final check (if it hangs) can you open the developer tools and see if any error messages show up in the console (Command Palette -> `Developer: Toggle Developer Tools` 
Thanks!

**Comment by gbockus-sf** [comment]
If you'd like to be more efficient with the back and forth debugging (and have the capacity) please join our support slack channel via this link and we'll get this figured out: https://join.slack.com/share/enQtNDc2NjI5MTgyMzIzNC0yNzMyZjYwMmQ0NGFiMDhmZjcyNjJhOWMzMGJmNzhlZmViNTU3NTE3MDExZmFjMjE3Y2IzZDJhZTcwNzVjZDhl

**Comment by martinlezer** [comment]
Hello @gbockus-sf,

Thanks for the quick answer.

Before raising the bug I tried the following steps:
- already authorized org
- new authorized org
- creation of new sfdx project
- changing sandbox as authorized org
- uninstall and reinstall all extensions
- switch to all compatible Java JDKs

I also tried all the steps you just provided and it's still not working with the last version of the Salesforce CLI extension.

You can find attached the logs provided by VS Code.
![Screenshot 2023-02-09 at 17 40 28](https://user-images.githubusercontent.com/112095125/217881151-6a8f8c21-cebe-4e1b-bdfb-23d59b79e8bd.png)

**Comment by gbockus-sf** [comment]
Thanks @martinlezer glad to help

![Screenshot 2023-02-09 at 10 53 24 AM](https://user-images.githubusercontent.com/76090802/217883079-e7d4eafd-717b-45d7-9312-68df92868732.png)
This one might be the culprit ( or it might be a completely unrelated extension).  I'll create a debug vsix with extra logging to narrow the issue down and share it. I'll target the retrieve via manifest flow just to limit the scope for now.

**Comment by gbockus-sf** [comment]
Howdy @martinlezer 
I updated a debug vsix file [here](https://github.com/gbockus-sf/salesforcedx-vscode/releases/tag/v56.16.0.1) Notes on the release on my fork explain how to install it and collect the logs.  
Thanks!

**Comment by martinlezer** [comment]
Hello @gbockus-sf,
Since neither retrieve nor push works, I collected the logs of both operations.
You can find them attached.
[vscode-logs.zip](https://github.com/forcedotcom/salesforcedx-vscode/files/10705154/vscode-logs.zip)

**Comment by martinlezer** [comment]
@gbockus-sf  new information. The run tests feature doesn't work even with the previous version of the CLI.

**Comment by gbockus-sf** [comment]
Hi @martinlezer 
I think I have it narrowed down to trying to find the correct API version to use when executing the retrieve.  Find a new debug vsix [here](https://github.com/gbockus-sf/salesforcedx-vscode/releases/download/v56.16.0.1/salesforcedx-vscode-core-57.2.0.vsix).  

Do you have the sourceApiVersion set in your sfdx-project.json file? 

The issue may have been addressed with some work done in this [PR](https://github.com/forcedotcom/salesforcedx-vscode/pull/4661).  Either way I'd be interested to see if any new error on surfaced in the logs for the debug vsix. 

Thanks!

**Comment by gbockus-sf** [comment]
This is still on my radar.  Basically the logs told me that it's going into the source-deploy-retrieve library and just hanging.  I'm working on the next debugging steps to figure out the issue.  Thanks for your patience.

**Comment by mayday91** [comment]
Same issue, same details, same OS and Processor. Still waiting on fix for this. Overall just cant retrieve or push any sources to or from org. It runs and just keeps running with no result and then fires problems/errors from not being able to communicate with org.

**Comment by mayday91** [comment]
Can I get a fresh link to the Slack support channel please? The one above has expired.

**Comment by agoganian** [comment]
I'm seeing the same issues. However, only recently (~1 week or so) -- not earlier (Feb) versions as others have reported.
* * * May be related to start of the issues * * *
A hard password reset preceded these problems
Following the reset I had to re-authorize my orgs, enter my ssh pass phrase for my corporate Github connection, regain ownership of documents in my local file system.

Symptoms (as described by others above)
sfdx cli works (deploy, change default org tested)
right-click diff or deploy do not work
right-click on manifest deploy/retrieve do not work
js console contains many of the same errors as posted previously by @martinlezer on Feb 9
  (node:76991) MaxListenersExceededWarning
  Failed to load local configuration: read ECONNRESET
  (node:76991) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security...

VSCODE:
Version: 1.77.3 (Universal)
Commit: 704ed70d4fd1c6bd6342c436f1ede30d1cff4710
Date: 2023-04-12T09:19:37.325Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Darwin arm64 22.4.0
Sandboxed: No

MacBook Pro
M1 Pro
16GB
macOS Ventura 13.3.1 (22E261)

**Comment by srosano** [comment]
I downloaded both the universal and apple silicon version of vscode - both didn't work. My current vscode is the apple silicon version of vscode as below: 

Version: 1.79.1
Commit: 4cb974a7aed77a74c7813bdccd99ee0d04901215
Date: 2023-06-12T16:17:16.971Z
Electron: 22.5.7
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Darwin arm64 22.5.0

Apple M1 Pro
MacOS Ventura 13.4 (22F66)

Not exactly sure what's happening but if I had to guess it might have something to do with how the previous vscode when it wasn't yet compatible by Apple Silicon might have to have been ran with Rosetta compatibility enabled or something similar

**Comment by diyer** [comment]
Closing this issue as there is no activity for more than 6 months. Please open a new issue for any new issue encountered in this area.
